toolkits/slack/arcade_slack/tools/chat.py

```diff
@@ -8,27 +7,371 @@
+    if conversation_id and channel_name:
+        raise ToolExecutionError("Provide exactly one of conversation_id OR channel_name.")
+
+    if not conversation_id:
+        channel = await get_conversation_metadata(
+            context=context,
+            channel_name=channel_name,
+        )
+        conversation_id = channel["id"]
+
+    auth_token = context.get_auth_token_or_empty()
+    slack_client = AsyncWebClient(token=auth_token)
+    user_ids, next_cursor = await async_paginate(
+        func=slack_client.conversations_members,
+        response_key="members",
+        limit=limit,
+        next_cursor=next_cursor,
+        channel=conversation_id,
+    )
+
+    users = await get_users_by_id(auth_token, user_ids)
+
+    raise_for_users_not_found(context, [users])
+
+    return {
+        "users": [user for user in users["users"] if not user.get("is_bot")],
+        "next_cursor": next_cursor,
+    }
+
+
+@tool(requires_auth=Slack(scopes=["mpim:history", "mpim:read", "users:read", "users:read.email"]))
+async def get_messages(
+    context: ToolContext,
+    conversation_id: Annotated[
+        str | None,
+        "The ID of the conversation to get messages from. Provide exactly one of conversation_id "
+        "OR any combination of user_ids, usernames, and/or emails.",
+    ] = None,
+    channel_name: Annotated[str | None, "The name of the channel to get messages from."] = None,
+    user_ids: Annotated[
+        list[str] | None, "The IDs of the users in the conversation to get messages from."
+    ] = None,
+    usernames: Annotated[
+        list[str] | None,
+        "The usernames of the users in the conversation to get messages from. Prefer providing"
+        "user_ids and/or emails, when available, since the performance is better.",
+    ] = None,
+    emails: Annotated[
+        list[str] | None,
+        "The emails of the users in the conversation to get messages from.",
+    ] = None,
+    oldest_relative: Annotated[
+        str | None,
+        (
+            "The oldest message to include in the results, specified as a time offset from the "
+            "current time in the format 'DD:HH:MM'"
+        ),
+    ] = None,
+    latest_relative: Annotated[
+        str | None,
+        (
+            "The latest message to include in the results, specified as a time offset from the "
+            "current time in the format 'DD:HH:MM'"
+        ),
+    ] = None,
+    oldest_datetime: Annotated[
+        str | None,
+        (
+            "The oldest message to include in the results, specified as a datetime object in the "
+            "format 'YYYY-MM-DD HH:MM:SS'"
+        ),
+    ] = None,
+    latest_datetime: Annotated[
+        str | None,
+        (
+            "The latest message to include in the results, specified as a datetime object in the "
+            "format 'YYYY-MM-DD HH:MM:SS'"
+        ),
+    ] = None,
+    limit: Annotated[int | None, "The maximum number of messages to return."] = None,
```

Comment:
can we limit to maybe 50 or 100 messages by default? I just asked to get dms between us & apparently we talk a lot b/c it timed out